### PR TITLE
Add server test suite and initial cases

### PR DESCRIPTION
**Pull request by saschagrunert**
This PR is a spin-off from #2002 to split it up into more review-able pieces. It mainly does:

- Add server test suite
- Add the `server/server.go` tests
- Add ginkgo table extension to vendor folder
- Add new mocked interfaces
- Add certificate test data
- Update streamService to be public

I know that "making things public for testing" is not a best practice, but this should really not harm, because the only way to access it is hidden via `server/server_test_inject.go`.

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+		})
+	})
+
+	t.Describe("StopStreamServer", func() {
+		// Prepare the sut
+		BeforeEach(setupSUT)
+
+		It("should succeed", func() {
+			// Given
+			/* gomock.InOrder(
```
Comment (mrunalp) [inline server/server_test.go]: Why is the commented out? Can we remove it if it is not needed?

Comment (saschagrunert) [reply server/server_test.go]: Yes, I will remove it.

**Pull request merged**